synapse/handlers/room.py

```diff
@@ -408,7 +409,16 @@ async def clone_existing_room(
             (EventTypes.ServerACL, ""),
             (EventTypes.RelatedGroups, ""),
             (EventTypes.PowerLevels, ""),
-        )
+        ]
+
+        # If the old room was a space, copy over the room type and the rooms in
+        # the space.
+        if (
+            old_room_create_event.content.get(EventContentFields.ROOM_TYPE)
+            == RoomTypes.SPACE
+        ):
+            creation_content[EventContentFields.ROOM_TYPE] = RoomTypes.SPACE
```

Comment:
Note that [MSC3385](https://github.com/matrix-org/matrix-doc/pull/3385) (made after this PR) suggests using the full create content. I think we can leave that to a follow-up depending on how that MSC goes...